rebasehelper/utils.py

```diff
@@ -227,34 +234,46 @@ def progress(download_total, downloaded, upload_total, uploaded):
             # no point to log progress, write directly to stdout
             sys.stdout.write('{:>3d}% {}\r'.format(int(r * 100), est))
             sys.stdout.flush()
-            return 0
 
-        if os.path.exists(destination_name):
-            return
-        with open(destination_name, 'wb') as f:
-            curl = pycurl.Curl()
-            curl.setopt(pycurl.URL, url)
-            curl.setopt(pycurl.CONNECTTIMEOUT, 30)
-            curl.setopt(pycurl.FOLLOWLOCATION, 1)
-            curl.setopt(pycurl.MAXREDIRS, 5)
-            curl.setopt(pycurl.TIMEOUT, 300)
-            curl.setopt(pycurl.WRITEDATA, f)
-            curl.setopt(pycurl.NOPROGRESS, 0)
-            curl.setopt(pycurl.PROGRESSFUNCTION, progress)
-            try:
+        try:
+            response = urllib.request.urlopen(url, timeout=timeout)
+            file_size = int(response.info().get('Content-Length', 0))
+
+            if not file_size:
+                raise DownloadError('The file has zero size')
+
+            # file exists, check the size
+            if os.path.exists(destination_path):
+                if file_size != os.path.getsize(destination_path):
+                    logger.debug("The destination file '%s' exists, but sizes don't match! Removing it.",
```

Comment:
This is only to decide if the existing file is correct or not. e.g. when you interrupt rebase-helper while it is downloading a file, the file will stay there, but the sizes will be different. We could improve it to delete the file in case the download didn't finish and there is KeyboardInterrupt, but I would leave that for different patch set.